coral/src/app/components/AsyncNativeSelectWrapper.tsx

```diff
@@ -0,0 +1,82 @@
+import { NativeSelect, NativeSelectProps, useToast } from "@aivenio/aquarium";
+import { isValidElement, ReactElement, ReactNode, useEffect } from "react";
+import { parseErrorMsg } from "src/services/mutation-utils";
+
+function isNativeSelectComponent(
+  child: ReactNode
+): child is ReactElement<NativeSelectProps> {
+  return (
+    isValidElement(child) &&
+    /* eslint-disable-next-line  @typescript-eslint/no-explicit-any */
+    (child.type as any)?.render?.displayName === NativeSelect.displayName
+  );
+}
+
+type AsyncNativeSelectWrapperProps = {
+  /**
+   * `entity` is the entity which user filters by, e.g. "Team" or "Topic"
+   * It is used in e.g. placeholder and toast notification for error cases,
+   * like "Error loading <ENTITY>"
+   */
+  entity: string;
+  isLoading: boolean;
+  isError: boolean;
+  error: unknown;
+  children: ReactElement<NativeSelectProps>;
+};
+
+/** <AsyncNativeSelectWrapper> handles loading
+ * and error states for a <NativeSelect> that uses
+ * data that is fetched async (e.g. from API)
+ * only takes one <NativeSelect /> as a child.
+ */
+function AsyncNativeSelectWrapper(props: AsyncNativeSelectWrapperProps) {
+  const { entity, isLoading, isError, error, children } = props;
+
+  const toast = useToast();
+
+  // Type-guard to make sure we're only passing
+  // <NativeSelect> as a child component
+  useEffect(() => {
+    if (!isNativeSelectComponent(children)) {
+      throw new Error(
+        "Invalid child component. `AsyncNativeSelectWrapper` only accepts `NativeSelect` as a child."
+      );
+    }
+  }, [children]);
+
+  useEffect(() => {
+    if (isError) {
+      toast({
+        message: `Error loading ${entity}: ${parseErrorMsg(error)}`,
+        position: "bottom-left",
+        variant: "default",
+      });
+    }
+  }, [isError]);
+
+  if (isError) {
+    return (
+      <NativeSelect
+        valid={false}
+        disabled={true}
+        labelText={`Filter by ${entity}`}
```

Comment:
Maybe "Filter by" is a bit too specific ? Maybe we can make the label `''`, or omit it entirely?

Reply:
True... "Filter by..." is what we usually use, but we can't be 100% sure that's always the case. It's so that the element has the same label as the valid NativeSelect, so there are no differences in layout.

Hm... this should match the labelText from NativeSelect and since this is a required property... I'll see if I can use that here! Then we are 100% sure that it always looks correct!

Reply:
worked! 🎉  https://github.com/Aiven-Open/klaw/pull/2176/commits/065b6b0c6833dbc5164d2e3f37483f9cbbf14560

Reply:
Very nice :o